logging.go

```diff
@@ -0,0 +1,27 @@
+package sgreplicate
+
+import (
+	"strings"
+	"github.com/couchbase/clog"
+	"fmt"
+)
+
+func LogTo(replication SGReplication, key string, format string, args ...interface{}) {
+	clog.To(key, prefixWithReplicationId(replication, format), args)
+}
+
+func Warn(replication SGReplication, args ...interface{}) {
+	if clog.Level <= clog.LevelWarning {
+		clog.Warn(prefixWithReplicationId(replication, fmt.Sprint(args...)))
+	}
+}
+
+func prefixWithReplicationId(replication SGReplication, raw string) string {
+	replicationId := replication.GetParameters().ReplicationId
+	if replicationId != "" {
+		return strings.Join([]string{"[",replicationId ,"] ", raw}, "")
+	}
+
+	return raw
+}
+
```

Comment:
That's the general idea - be able to call r.LogTo instead of LogTo(r), to make it clearer that LogTo and Warn are expected to get called within the context of a replication.  

I take your point that it's going to be a bit more refactoring to implement those methods in both types of replicator, and if they just end up calling common code anyway, it's not necessarily making things cleaner.  I wonder what the idiomatic go way would be to implement this - define a LoggingReplicator type that implements both methods, and add that to each of the replicator types?

I'm fine with leaving as-is if you think the current implementation is clear enough.